common/persistence/class.PhpRedisDriver.php

```diff
@@ -189,30 +192,94 @@ public function hGet($key, $field)
 
     public function hDel($key, $field): bool
     {
-        return (bool) $this->callWithRetry('hDel', [$key, $field]);
+        return (bool)$this->callWithRetry('hDel', [$key, $field]);
     }
 
     //Time complexity: O(N) where N is the size of the hash.
     public function hGetAll($key)
     {
         return $this->callWithRetry('hGetAll', [$key]);
     }
+
     //Time complexity: O(N)
     public function keys($pattern)
     {
         return $this->callWithRetry('keys', [$pattern]);
     }
+
     //Time complexity: O(1)
     public function incr($key)
     {
         return $this->callWithRetry('incr', [$key]);
     }
+
     //Time complexity: O(1)
     public function decr($key)
     {
         return $this->callWithRetry('decr', [$key]);
     }
 
+    /**
+     * @throws RedisException
+     * @throws common_exception_Error
+     */
+    public function scan(int &$iterator = null, string $pattern = null, int $count = 1000): array
+    {
+        $retry = $this->params['attempt'];
+        $attempt = 0;
+
+        while ($attempt <= $retry) {
+            try {
+                return $this->connection->scan($iterator, $pattern, $count);
+            } catch (Exception $exception) {
+                common_Logger::d('Redis  ' . $method . ' failed ' . $attempt . '/' . $retry . ' :  ' . $exception->getMessage());
+
+                if ($e->getMessage() == 'Failed to AUTH connection' && isset($this->params['password'])) {
```

Comment:
'Failed to AUTH connection' a constant for this would be nice :)

Reply:
I'd maybe add a comment saying why we compare against a exception message (i.e. because (I think) the extension always uses the same exception class and there is no exception code we can use to get why it failed).